### ISSUE TITLE
Feature: Support "windows terminal" in addition to mintty and cmd

### ISSUE THREAD

**Issue by Diggsey** [git-for-windows/git#2621]
When installing I'm presented with the option to use either `mintty` or `cmd` with `git bash`. [Windows terminal](https://github.com/microsoft/terminal) would be a good option to add going forward as it is officially supported by Microsoft, but is not... like `cmd`.

**Comment by dscho** [comment]
I like the idea, but there are challenges with it:

- We cannot bundle Windows Terminal, so we'd have to somehow detect its presence
- Last time I checked, there were some funny issues when resizing the window, and that's when I stopped checking whether Windows Terminal is compatible with Git for Windows. Admittedly, this has been a long time ago.
- The Git for Windows project is not really in need of more feature requests, but in need of more contributors taking care of such feature requests. I really hope that you turn out to be such a contributor.

In short: I would be delighted to assist you in your endeavors to make this happen.

**Comment by Diggsey** [comment]
It's a windows store app so if the user selects that option it should be fairly straightforward to automatically install that.

**Comment by dscho** [comment]
> It's a windows store app so if the user selects that option it should be fairly straightforward to automatically install that.

Yep. Go right ahead implementing that.

**Comment by lhotamir** [comment]
This would be a great feature. I already use git bash with windows terminal, but I cannot get it to work in the xterm-256color mode. I cannot understand why, since git bash works in mintty with this mode, and for example cygwin bash works in windows terminal with this mode too.

**Comment by dscho** [comment]
@lhotamir so how about helping out? This feature request won't implement itself...

**Comment by noinkling** [comment]
Just for reference:

https://docs.microsoft.com/en-us/windows/terminal/command-line-arguments

`wt.exe` is made available in the `PATH` when installed, for me it's located at `%LOCALAPPDATA%\Microsoft\WindowsApps\wt.exe`

It seems to always open a new window, rather than adding a tab to an existing one. I tried resource hacking the command string for `git-bash.exe` (still using `usr/bin/bash.exe`), which launched it ok, but the environment variables weren't set correctly, e.g. `MSYSTEM`. They are set when `bin/bash.exe` instead however.

On the Windows Terminal side there's an effort to allow third party programs to add/modify profiles, which might be useful in the future: https://github.com/microsoft/terminal/blob/main/doc/specs/Proto%20extensions-spec.md

There's also a request to add a profile for Git Bash by default if it's installed, similar to what already happens with WSL: microsoft/terminal#1394

**Comment by linquize** [comment]
I suggest in Git installer, add a checkbox **add Git to Windows terminal** and append the new entry to Windows Terminal `profiles.json`

**Comment by fourpastmidnight** [comment]
The last time I added G4w bash to Windows Terminal, it mostly worked, but not everything. Anyway, it was enough that I opted to remove it from Windows Terminal and keep using mintty.

Unless something has significantly changed, I think maybe this shouldn't be part of the official G4w installer, as it may become a steady source of issues which will become a distraction to the g4w team.

At most, I might provide instructions on the Readme for adding g4w to the Windows Terminal with a disclaimer that g4w will not support any issues from running it in Windows Terminal.

**Comment by noinkling** [comment]
I'll just say that Windows Terminal has been steadily improving, and it's intended to eventually supersede conhost as the default terminal on Windows. I think as time progresses you'll see more people looking for a quick/easy way to use Git Bash with it, even if it's just providing a profile on install if/when that RFC that I linked above comes to fruition.

**Comment by dscho** [comment]
@noinkling I am definitely not opposed to including this as an experimental feature in the Git for Windows installer. It's just that _somebody_ has to do the work, and that somebody isn't me. If that somebody is you, here are a couple of pointers:

- https://github.com/git-for-windows/git/wiki/Making-an-installer describes how you can build a Git for Windows installer locally
- https://github.com/git-for-windows/build-extra/blob/master/installer/install.iss is the file defining the behavior of Git for Windows' installer
- [This part](https://github.com/git-for-windows/build-extra/blob/8a8198a9391975d508dbb14378517979770a6ef7/installer/install.iss#L2858-L2864) of the installer is responsible for (optionally) using ConHost instead of MinTTY for the Git Bash.

Feel free to ask questions!

**Comment by Okeanos** [comment]
@dscho based on recent releases of Git that include the Windows Terminal Fragment (#3183) is this still relevant?

**Comment by dscho** [comment]
Good point! Thanks @Okeanos for paying attention.
